wayang-docs/src/main/resources/how_contribute/code_changes/preparing_contribute_code_changes/branching_strategy.md

```diff
@@ -0,0 +1,58 @@
+---
+license: |
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+    
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+layout: default
+title: "Branching Strategy"
+previous:
+    url: /how_contribute/code_changes/code_review_criteria/
+    title: Code Review Criteria
+next:
+    url: /how_contribute/code_changes/preparing_contribute_code_changes/jira_issue/
+    title: Jira issues
+---
+# Apache Wayang-Branching Strategy
+This page is a summary from an email thread: https://lists.apache.org/thread/v9nhw3ml585456byd6t6cbn3vbhnzkww
+
+## Branches:
+__*release (formerly main):*__ 
+
+This branch represents the current stable state of Wayang. It's where users can find the latest and most stable version of the project. The content of this branch is the same as the latest "rel/x.x.x" branch.
+
+
+__*rel/x.x.x (Release Version Branches):*__
+
+These branches are used to track all versions of Wayang. Older versions are kept here for reference and use, similar to tags. For example, if you need to use an older version of Wayang, you can find it here.
+
+
+__*cherry-pick (Feature/Fix Selection):*__
+
+This branch is used when selecting specific features and fixes implemented on the "develop" branch to be included in the next release. It allows for testing and ensuring compatibility with Apache guidelines before creating the "rel/" branch.
+
+
+__*develop (Development Branch):*__
```

Comment:
After some consideration, we will use the "main" branch to be the one where we will merge our PRs and open new branches.